srcf/controllib/jobs.py

```diff
@@ -95,10 +100,11 @@ def subproc_call(job, desc, cmd, stdin=None):
     if out:
         try:
             out = out.decode("utf-8")
-        except UnicodeDecodeError as e:
-            output = b"[Could not decode output as UTF-8]\n" + output
+        except UnicodeDecodeError:
+            out = b"[Could not decode output as UTF-8]\n" + out
         job.log(desc, "output", raw=out)
```

Comment:
Is it right to pass a `bytes` to Job.log()?  The field in the database is of `text` type, which stores characters (not bytes).  I don't know what Postgres does if _it_ encounters invalid UTF-8 bytes, but we should find out.

(Side note: can't work out what the `self.logger` in Job.log() actually refers to -- does this somehow get the module-global logger?  I thought that wouldn't be accessible as a member of `self`...)

Reply:
PostgreSQL seems to be ok with storing random bytes in a `text` field (and psql will just display it in hex), but the control panel frontend is less keen on it -- looks like [we have some handling for this already](https://github.com/SRCF/control-panel/blob/65c2f3577b48f003058196596ef117fe1b986bce/control/templates/admin/status.html#L101) but it assumes well-formed unicode.  I've added `JobLog.raw_safe` here and will update the frontend to use it post-merge.

(The job runner [assigns its own logger](https://github.com/SRCF/srcf-python/blob/0cf9ed8755c9a74acfbd81b40084518fc3060e2b/srcf/controllib/job_runner.py#L174) to `job.logger` for each job as it passes through.)